core/src/main/scala/org/apache/spark/shuffle/FileShuffleBlockManager.scala

```diff
@@ -133,7 +134,8 @@ class FileShuffleBlockManager(conf: SparkConf)
               logWarning(s"Failed to remove existing shuffle file $blockFile")
             }
           }
-          blockManager.getDiskWriter(blockId, blockFile, serializer, bufferSize, writeMetrics)
+          blockManager.getDiskWriter(blockId, blockFile, serializerInstance, bufferSize,
```

Comment:
Note that this line is called one for every bucket (reduce task), since it's enclosed in

```
Array.tabulate[BlockObjectWriter](numBuckets) { bucketId =>
...
```